pgjdbc/src/main/java/org/postgresql/util/PasswordUtil.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2023, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+import org.postgresql.core.Utils;
+
+import com.ongres.scram.common.ScramFunctions;
+import com.ongres.scram.common.ScramMechanisms;
+import com.ongres.scram.common.bouncycastle.base64.Base64;
+import com.ongres.scram.common.stringprep.StringPreparations;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.sql.SQLException;
+
+public class PasswordUtil {
+  private static final String DEFAULT_PASSWORD_ENCRYPTION = "scram-sha-256";
+  private static final int DEFAULT_ITERATIONS = 4096;
+  private static final int DEFAULT_SALT_LENGTH = 16;
+
+  /**
+   * Generate the encoded text representation of the given password for
+   * SCRAM-SHA-256 authentication. The return value of this method is the literal
+   * text that may be used when creating or modifying a user with the given
+   * password without the surrounding single quotes.
+   *
+   * @param password   The plain text of the user's password
+   * @param iterations The number of iterations of the hashing algorithm to
+   *                   perform
+   * @param salt       The random salt value
+   * @return T text representation of the password encrypted for SCRAM-=SHA-256
+   *         authentication
+   */
+  public static String encodeScramSha256(String password, int iterations, byte[] salt) {
+    if (iterations <= 0) {
+      throw new IllegalArgumentException("iterations must be greater than zero");
+    }
+    if (salt.length == 0) {
+      throw new IllegalArgumentException("salt length must be greater than zero");
+    }
+    byte[] saltedPassword = ScramFunctions.saltedPassword(ScramMechanisms.SCRAM_SHA_256,
+        StringPreparations.SASL_PREPARATION, password, salt, iterations);
+    byte[] clientKey = ScramFunctions.clientKey(ScramMechanisms.SCRAM_SHA_256, saltedPassword);
+    byte[] storedKey = ScramFunctions.storedKey(ScramMechanisms.SCRAM_SHA_256, clientKey);
+    byte[] serverKey = ScramFunctions.serverKey(ScramMechanisms.SCRAM_SHA_256, saltedPassword);
+
+    StringBuilder sb = new StringBuilder("SCRAM-SHA-256") //
```

Comment:
Could you use string concatenation here? There's no control flow, so no need in `StringBuilder`